File: src/algorithms/is-ugly.js
/**
 * An ugly number is a positive integer whose prime factors are limited to
 * 2, 3, and 5. Given an integer n, returns true if n is an ugly number.
 * 
 * @param {number} n - A number parameter
 * @returns {boolean} - Returns true if n is an ugly number
 */
function isUgly (n) {
  if (n < 1) {
    return false;
  }
  let primes = [2, 3, 5];
  for (const prime of primes) {
    while (n % prime === 0) {
      n /= prime;
    }
  }
  return n === 1;
}

File: src/algorithms/is-ugly.py
def isUgly(n):
    """An ugly number is a positive integer which does not have a prime factor 
    other than 2, 3, and 5. Given an integer n, return true if n is an ugly 
    number.

    Args:
        n (int): An integer. 

    Returns:
        Boolean: Returns True if n is an ugly number.
    """
    if n < 1:
        return False
    primes = [2, 3, 5]
    for prime in primes:
        while n % prime == 0:
            n /= prime
    return n == 1
